### PR TITLE
Added port command instead for searching for ports

### PR DESCRIPTION
**Pull request by fayeed**

### CODE REVIEW

File: src-tauri/src/main.rs [hunk 1]
```diff
@@ -24,13 +24,21 @@ use directories::UserDirs;
...
 #[derive(Clone, serde::Serialize)]
 struct Payload {
   message: String,
 }
 
+pub struct FreePort(pub u16);
+
+#[tauri::command]
+fn get_ad4m_port(state: State<'_, FreePort>) -> u16 {
+    state.0.clone()
```
Comment (kaichaosun) [inline src-tauri/src/main.rs]: You don't need clone here.

File: src/context/Ad4minContext.tsx [hunk 2]
```diff
@@ -128,97 +123,59 @@ export function Ad4minProvider({ children }: any) {
...
-            message: 'Cannot connect to the URL provided please check if the executor is running or pass a different URL',
-            color: 'red',
-            autoClose: false
-          });
-        });
+        connect(localStorageURL);
       }
     } else {
-      findAd4mPort();
+      invoke('get_ad4m_port').then((message) => {
```
Comment (kaichaosun) [inline src/context/Ad4minContext.tsx]: How about use name `get_port` or `get_graphql_port`?

Also, can you use await syntax?

File: src/context/Ad4minContext.tsx [hunk 3]
```diff
@@ -128,97 +123,59 @@ export function Ad4minProvider({ children }: any) {
...
-          signal: Timeout().signal
-        })
-        const client = buildAd4mClient(url);
-        const ad4mUrl = await checkConnection(url, client);
-
-        const {isInitialized, isUnlocked} = await checkIfAgentIsInitialized(client);
 
+  const connect = (url: string) => {
+    const client = buildAd4mClient(url);
+    checkConnection(url, client).then((url) => {
```
Comment (kaichaosun) [inline src/context/Ad4minContext.tsx]: Also here use await syntax.

File: src/context/Ad4minContext.tsx (hunk 2 above)
Comment (fayeed) [reply src/context/Ad4minContext.tsx]: Calling this inside `useEffect` so async-await doesn't work

Comment (fayeed) [reply src/context/Ad4minContext.tsx]: using `get_port` as command name.

File: src/context/Ad4minContext.tsx [hunk 4]
```diff
@@ -128,97 +123,63 @@ export function Ad4minProvider({ children }: any) {
...
+        }
+      })
     }
-  }, [checkConnection, checkIfAgentIsInitialized, findAd4mPort]);
+  }, [checkConnection, checkIfAgentIsInitialized]);
 
   useEffect(() => {
     appWindow.listen('ready', () => {
-      findAd4mPort();
+      invoke('get_port').then((message) => {
```
Comment (kaichaosun) [inline src/context/Ad4minContext.tsx]: Better use await here.

File: src-tauri/src/main.rs [hunk 5]
```diff
@@ -75,7 +85,7 @@ fn main() {
 
             let (mut rx, _child) = Command::new_sidecar("ad4m")
             .expect("Failed to create ad4m command")
-            .args(["serve", "--port", &free_port.to_string()])
+            .args(["serve", "--port", &free_port_clone.to_string()])
```
Comment (kaichaosun) [inline src-tauri/src/main.rs]: Inline `free_port_clone` or maybe don't need to clone at all?

**Review by kaichaosun — APPROVED** [review]

**Pull request merged**